From 09219bf8efe5051318a7ec27ceba64144b728189 Mon Sep 17 00:00:00 2001
From: "Du, Changbin" <changbin.du@intel.com>
Date: Mon, 27 Apr 2015 18:54:17 +0800
Subject: [PATCH] usb/dwc3: do not suspend gadget for u1 and u2 state

U1 and U2 are two low power link state, U3 is the link state where a
device is put into a suspend state. Do not suspend gadget until link
is put into suspend state.

Change-Id: Ie3424c49c88eb477c695edec086ca93576e89f0c
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-30602
Signed-off-by: Du, Changbin <changbin.du@intel.com>
---
 drivers/usb/dwc3/gadget.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/drivers/usb/dwc3/gadget.c b/drivers/usb/dwc3/gadget.c
index 1fcc493e0a62..7c5f4a49a06a 100644
--- a/drivers/usb/dwc3/gadget.c
+++ b/drivers/usb/dwc3/gadget.c
@@ -2585,11 +2585,6 @@ static void dwc3_gadget_linksts_change_interrupt(struct dwc3 *dwc,
 	}
 
 	switch (next) {
-	case DWC3_LINK_STATE_U1:
-		if (dwc->speed == USB_SPEED_SUPER)
-			dwc3_suspend_gadget(dwc);
-		break;
-	case DWC3_LINK_STATE_U2:
 	case DWC3_LINK_STATE_U3:
 		dwc3_suspend_gadget(dwc);
 		break;
-- 
1.9.1

